python/cuml/naive_bayes/naive_bayes.py

```diff
@@ -0,0 +1,533 @@
+        """
+        Incremental fit on a batch of samples.
+
+        This method is expected to be called several times consecutively on
+        different chunks of a dataset so as to implement out-of-core or online
+        learning.
+
+        This is especially useful when the whole dataset is too big to fit in
+        memory at once.
+
+        This method has some performance overhead hence it is better to call
+        partial_fit on chunks of data that are as large as possible (as long
+        as fitting in the memory budget) to hide the overhead.
+
+        Parameters
+        ----------
+
+        X : {array-like, cupy sparse matrix} of shape (n_samples, n_features)
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features
+
+        y : array-like of shape (n_samples) Target values.
+        classes : array-like of shape (n_classes)
+                  List of all the classes that can possibly appear in the y
+                  vector. Must be provided at the first call to partial_fit,
+                  can be omitted in subsequent calls.
+
+        sample_weight : array-like of shape (n_samples)
+                        Weights applied to individual samples (1. for
+                        unweighted). Currently sample weight is ignored
+
+        Returns
+        -------
+
+        self : object
+        """
+        return self._partial_fit(X, y, sample_weight=sample_weight,
+                                 _classes=classes)
+
+    @cp.prof.TimeRangeDecorator(message="predict()", color_id=1)
+    def predict(self, X):
+
+        """
+        Perform classification on an array of test vectors X.
+
+        Parameters
+        ----------
+
+        X : array-like of shape (n_samples, n_features)
+
+        Returns
+        -------
+
+        C : cupy.ndarray of shape (n_samples)
+
+        """
+
+        jll = self._joint_log_likelihood(X)
+        indices = cp.argmax(jll, axis=1).astype(self.classes_.dtype)
+
+        y_hat = invert_labels(indices, classes=self.classes_)
+        return y_hat
+
+    def predict_log_proba(self, X):
+
+        """
+        Return log-probability estimates for the test vector X.
+
+        Parameters
+        ----------
+
+        X : array-like of shape (n_samples, n_features)
+
+
+        Returns
+        -------
+
+        C : array-like of shape (n_samples, n_classes)
+            Returns the log-probability of the samples for each class in the
+            model. The columns correspond to the classes in sorted order, as
+            they appear in the attribute classes_.
+        """
+        jll = self._joint_log_likelihood(X)
+
+        # normalize by P(X) = P(f_1, ..., f_n)
+
+        # Compute log(sum(exp()))
+
+        # Subtract max in exp to prevent inf
+        a_max = cp.amax(jll, axis=1, keepdims=True)
+
+        logsumexp = cp.log(cp.sum(cp.exp(jll - a_max), axis=1))
+
+        a_max = cp.squeeze(a_max, axis=1)
```

Comment:
Interesting. Do these not inherit the allocator from the ndarrays they are wrapping?

Reply:
Indeed, it looks like the allocator is not being inherited. What are your thoughts on having one of our `__init__.py` files call `cp.cuda.set_allocator(rmm.rmm_cupy_allocator)`? This might alleviate the need to have to do it manually everywhere.